Formula/aws-cdk.rb

```diff
@@ -0,0 +1,20 @@
+require "language/node"
+
+class AwsCdk < Formula
+  desc "AWS Cloud Development Kit - framework for defining AWS infra as code"
+  homepage "https://github.com/aws/aws-cdk"
+  url "https://registry.npmjs.org/aws-cdk/-/aws-cdk-1.0.0.tgz"
+  sha256 "3cc46b40fa5fa7134ca86a9caeab3d16895d13d503cd8f839861ade96056ff64"
+
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    output = shell_output("#{bin}/cdk --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Apologies, I've added a (hopefully) valid test of functionality that doesn't require creds. My poor defence is that I was looking at other formula for inspiration and some of what I saw were version checks.